### ISSUE TITLE
Failed to install Ruby Stable on (Ubuntu 22.04)

### ISSUE THREAD

**Issue by alilland** [rvm/rvm#5219]
## Description

Failed to install `\curl -sSL https://get.rvm.io | bash -s stable --ruby`
 it got almost to the end of compile, and it told me to look at the make.log file for details

there were a lot of warnings listed, but and one error --
```
ossl_pkey_rsa.c:885:58: error: ‘RSA_SSLV23_PADDING’ undeclared (first use in this function); did you mean ‘RSA_NO_PADDING’?
  885 | #define DefRSAConst(x) rb_define_const(cRSA, #x, INT2NUM(RSA_##x))
      |                                                          ^~~~
ossl_pkey_rsa.c:950:5: note: in expansion of macro ‘DefRSAConst’
  950 |     DefRSAConst(SSLV23_PADDING);
      |     ^~~~~~~~~~~
ossl_pkey_rsa.c:885:58: note: each undeclared identifier is reported only once for each function it appears in
  885 | #define DefRSAConst(x) rb_define_const(cRSA, #x, INT2NUM(RSA_##x))
      |                                                          ^~~~
ossl_pkey_rsa.c:950:5: note: in expansion of macro ‘DefRSAConst’
  950 |     DefRSAConst(SSLV23_PADDING);
      |     ^~~~~~~~~~~
ossl_pkey_rsa.c: At top level:
cc1: note: unrecognized command-line option ‘-Wno-self-assign’ may have been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-parentheses-equality’ may have been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option ‘-Wno-constant-logical-operand’ may have been intended to silence earlier diagnostics
make[2]: *** [Makefile:328: ossl_pkey_rsa.o] Error 1
make[2]: Leaving directory '/home/aron/.rvm/src/ruby-3.0.0/ext/openssl'
make[1]: *** [exts.mk:260: ext/openssl/all] Error 2
make[1]: Leaving directory '/home/aron/.rvm/src/ruby-3.0.0'
make: *** [uncommon.mk:300: build-ext] Error 2
++ /scripts/functions/support : __rvm_make()  383 > return 2

```

## Steps to reproduce

1) `sudo apt update`
2) `sudo apt install gnupg2 build-essential git libssl-dev libreadline-dev bison zlib1g-dev autoconf libgdbm-dev libyaml-dev libreadline-dev libncurses5-dev libffi-dev`
3) `gpg2 --keyserver hkp://keyserver.ubuntu.com --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 7D2BAF1CF37B13E2069D6956105BD0E739499BDB`
4) `\curl -sSL https://get.rvm.io | bash -s stable --ruby`

## Environment info

Using a clean install of Ubuntu 22.04 with GUI

**Comment by raelgc** [comment]
This is a duplicate of #5209.

Anyway,  if you're willed to test my compiled rubies, please try the following:

```terminal
cd /usr/share/rvm/rubies
wget https://raelcunha.com/share/ruby-2.7.6.tar.bz2
tar -xjvf ruby-2.7.6.tar.bz2
rvm mount /usr/share/rvm/rubies/ruby-2.7.6/
```

If you're using default `rvm` (installed manually):
```terminal
cd ~/.rvm/rubies
wget https://raelcunha.com/share/ruby-2.7.6.tar.bz2
tar -xjvf ruby-2.7.6.tar.bz2
rvm ~./rvm/rubies/ruby-2.7.6/
```

There is a `3.0.4` binary too, just replace the `2.7.6` in the above commands. 

Please, give feedback. If it works, I'll publish as compiled binaries.

**Comment by alilland** [comment]
@raelgc i'll keep an eye out on the progress of https://github.com/rvm/rvm/issues/5209 
I was just messing around with a VirtualBox VM preparing for a meeting later on in the week, fortunately I dont have anything business critical at this time 😅  

Appreciate it though

**Comment by davidkrider** [comment]
hq785@miner:~$ rvm use 3.0.4
Gemset '' does not exist, 'rvm ruby-3.0.4 do rvm gemset create ' first, or append '--create'.
hq785@miner:~$ rvm ruby-3.0.4 do rvm gemset create
Ruby ruby-3.0.4 is not installed.
